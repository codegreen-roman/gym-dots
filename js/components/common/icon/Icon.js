import React from 'react'
import { string, number } from 'prop-types'
import { svgStyle, pathStyle } from './Icon.glamor'

const icons = {
    facebook:
        'M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z'
}

export const Icon = ({ icon, size, color }) => {
    return (
        <svg
            {...svgStyle}
            width={`${size}px`}
            height={`${size}px`}
            viewBox='0 0 1024 1024'
        >
            <path {...pathStyle(color)} d={icons[icon]} />
        </svg>
    )
}

Icon.propTypes = {
    icon: string.isRequired,
    size: number,
    color: string
}

Icon.defaultProps = {
    size: 16
}
