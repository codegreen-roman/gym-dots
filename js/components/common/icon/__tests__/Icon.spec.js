import React from 'react'
import { render } from 'enzyme'
import { Icon } from '../Icon'

describe('Icon Component', () => {
    it('renders with defaults', () => {
        const wrapper = render(<Icon icon='facebook' color='grey' />)
        expect(wrapper).toMatchSnapshotWithGlamor()
    })
})
