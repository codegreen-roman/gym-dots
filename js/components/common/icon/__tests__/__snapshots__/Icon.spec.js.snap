// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders with defaults 1`] = `
<svg
  data-css-4t1vl3=""
  height="16px"
  viewbox="0 0 1024 1024"
  width="16px"
>
  <path
    d="M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z"
    data-css-rn73cr=""
  />
</svg>
`;
